Userland/Libraries/LibWeb/DOM/Node.h

```diff
@@ -162,6 +163,9 @@ class Node
 
     bool is_host_including_inclusive_ancestor_of(const Node&) const;
 
+    // Used for dumping the DOM Tree
+    void dump_tree_json(JsonObjectSerializer<StringBuilder>&) const;
```

Comment:
Maybe call this `serialize_tree_as_json` (or just `serialize_tree`) or something like this, then it's maybe a bit clearer what this does.